mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -51,6 +58,17 @@
 
     forwardConfigSettings(job);
 
+    //Convert Hive filter to Iceberg filter
+    String hiveFilter = job.get(TableScanDesc.FILTER_EXPR_CONF_STR);
```

Comment:
I don't think `getSplits` is the right method to set the translated Iceberg expr. I think this should go in `IcebergStorageHandler` . @guilload what do you think?

Reply:
Yeah I think you're right here, and I was initially attempting to do it that way too but I was having some trouble finding the correct method of doing that. I attempted to update the conf in `decomposePredicate` but the way I was doing it caused some bugs where the value I was setting wasn't getting reset between queries if one query had a filter (and decomposePredicate would get called) and the next didn't (so decomposePredicate doesn't get called). And in the other methods like `configureJobConf`, the filter information wasn't available so it couldn't be set. 

I would really appreciate some input on this, I may have missed something with how the StorageHandler works with Hive!

Reply:
@omalley, any input on where the filter pushdown should go?

Reply:
The most common places to put it are:
* getSplits - to filter the splits aggressively
* getRecordReader - to filter the records while reading

The information is being set by HiveInputFormat.pushFilters, but looking through the various paths to get there I think you need to stick to those.

Reply:
I came to the same conclusion when trying to implement projection pushdown in the storage handler.

Unfortunately as @cmathiesen stated, the job config is not yet populated with the projected columns and the filter expression when the storage handler "hooks" such as `configureJobConf` are called. So the right entry points for implementing PPD are `getSplits` and  `getRecordReader`.

However, there's another catch. The `JobConf` objects passed in `getSplits` and `getRecordReader` are actually not the same and the filter expression set in `getSplits` (L#53 in `HiveIcebergInputFormat`)  is no longer available when `getRecordReader ` is subsequently called.

Since in the storage handler we don't decompose the filter expression, Hive applies the whole thing anyway and this can't be caught in the test suite but we need to set the filter expression both in `getSplits` and `getRecordReader` to get to a complete PPD implementation.

Reply:
Can't the record reader use the residuals that are present in each task produced by `getSplits`? I don't think we need to get the filter twice as long as `getSplits` is using it and creating tasks with residuals.

Reply:
Oh very good point. Tasks do carry the residual filter.

Reply:
and looking at the code, the underlying MR input format already does the right thing and passes the residual filter from the task down to the reader. Great! My comment is only relevant for projection pushdown then. My bad.

Reply:
That makes sense. Seems like we cannot do this for predicate and projection pushdown in StorageHandler itself.